modules/core/src/main/scala/lucuma/odb/api/model/ConstraintSetModel.scala

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.api.model
+
+import cats._
+import cats.data.State
+import cats.syntax.all._
+import clue.data.Input
+import eu.timepit.refined.types.string.NonEmptyString
+import io.circe.Decoder
+import io.circe.generic.semiauto._
+import lucuma.core.enum._
+import lucuma.core.model.{ ConstraintSet, Program }
+import lucuma.core.optics.syntax.lens._
+import lucuma.odb.api.model.Existence._
+import lucuma.odb.api.model.syntax.input._
+import monocle.{ Fold, Lens, Optional }
+import monocle.macros.GenLens
+
+final case class ConstraintSetModel(
+  id:              ConstraintSet.Id,
+  existence:       Existence,
+  programId:       Program.Id,
```

Comment:
Having a program id here means that unlike targets, we don't anticipate sharing constraint sets across programs.  I agree that's a fine assumption to start with.

Reply:
Yeah. I guess it was an assumption that they wouldn't be shared between programs. I can ask Andy, though.

Reply:
Andy has stated that constraint sets won't be shared across programs.

Reply:
Sounds good.